packages/app/src/components/inspector/LineInspector.tsx

```diff
@@ -0,0 +1,189 @@
+import { useSelector } from 'noya-app-state-context';
+import { Button, Spacer } from 'noya-designsystem';
+import { decodeCurvePoint, Point, Selectors, SetNumberMode } from 'noya-state';
+import { isPointsLayer, clampRotation } from 'noya-state';
+import { useCallback, useMemo } from 'react';
+import styled, { useTheme } from 'styled-components';
+import LinePointCoordinatesInspector from '../../containers/LinePointCoordinatesInspector';
+import useShallowArray from '../../hooks/useShallowArray';
+import FlipHorizontalIcon from '../icons/FlipHorizontalIcon';
+import FlipVerticalIcon from '../icons/FlipVerticalIcon';
+import DimensionInput from './DimensionInput';
+
+export type DimensionValue = number | undefined;
+
+const Row = styled.div(({ theme }) => ({
+  flex: '0 0 auto',
+  display: 'flex',
+  flexDirection: 'row',
+  paddingLeft: '10px',
+  paddingRight: '10px',
+}));
+
+const FlipButtonContainer = styled.div(({ theme }) => ({
+  flex: '1',
+  display: 'flex',
+  flexDirection: 'row',
+}));
+
+const LinePointCoordinatesContainer = styled.div(({ theme }) => ({
+  width: '200px',
+}));
+
+function roundNumber(number: number, roundTo: number) {
+  return Number(number.toFixed(roundTo));
+}
+
+export interface Props {
+  width: DimensionValue;
+  isFlippedVertical: boolean;
+  isFlippedHorizontal: boolean;
+  rotation: DimensionValue;
+  hasLineLayer: boolean;
+  onSetWidth: (value: number, mode: SetNumberMode) => void;
+  onSetHeight: (value: number, mode: SetNumberMode) => void;
+  onSetRotation: (value: number, mode: SetNumberMode) => void;
+  onSetIsFlippedVertical: (value: boolean) => void;
+  onSetIsFlippedHorizontal: (value: boolean) => void;
+}
+
+export default function LineInspector({
+  width,
+  rotation,
+  isFlippedVertical,
+  isFlippedHorizontal,
+  hasLineLayer,
+  onSetWidth,
+  onSetRotation,
+  onSetIsFlippedVertical,
+  onSetIsFlippedHorizontal,
+}: Props) {
+  const {
+    icon: iconColor,
+    iconSelected: iconSelectedColor,
+  } = useTheme().colors;
+
+  const handleSetIsFlippedVertical = useCallback(
+    () => onSetIsFlippedVertical(!isFlippedVertical),
+    [isFlippedVertical, onSetIsFlippedVertical],
+  );
+
+  const handleSetIsFlippedHorizontal = useCallback(
+    () => onSetIsFlippedHorizontal(!isFlippedHorizontal),
+    [isFlippedHorizontal, onSetIsFlippedHorizontal],
+  );
+
+  const flipButtonElements = useMemo(
```

Comment:
We should make a new component for these flip controls so we don't have to duplicate them